### PR TITLE
Feat/add ES query option to lock command

### PR DESCRIPTION
**Pull request by coppee**
|Q              |A  |
|---------------|---|
|Bug fix?       ||
|New feature?   |Y|	
|BC breaks?     ||	
|Deprecations?  ||	
|Fixed tickets  ||	

Big picture: I need to get the ES query feature in the Recompute command. But this one use Lock command in his logic, so first I need to implement this feature here. I also "upgraded" the look of this class a bit (with a "little" inspiration from the BulkActionCommand)

### CODE REVIEW

**Comment by Davidmattei** [comment]
You should create more PR's! Clean commits -> easy review +1

**Review by Davidmattei — APPROVED** [review]

File: src/Command/LockCommand.php [hunk 1]
```diff
@@ -12,23 +18,33 @@
...
-     */
-    private $contentTypeRepository;
+    private string $by;
+    private ContentType $contentType;
+    private ContentTypeRepository $contentTypeRepository;
+    private ElasticaService $elasticaService;
+    private bool $force;
+    private SymfonyStyle $io;
+    /** @var mixed */
+    private $query;
```
Comment (theus77) [inline src/Command/LockCommand.php:30]: Is it not a string?

File: src/Command/LockCommand.php [hunk 2]
```diff
@@ -37,64 +53,121 @@ protected function configure(): void
...
             ->setDescription('Lock a content type')
-            ->addArgument('contentType', InputArgument::REQUIRED, 'content type to recompute')
-            ->addArgument('time', InputArgument::REQUIRED, 'lock until (+1day, +5min, now)')
-            ->addOption('user', null, InputOption::VALUE_REQUIRED, 'lock username', 'EMS_COMMAND')
-            ->addOption('force', null, InputOption::VALUE_NONE, 'do not check for already locked revisions')
-            ->addOption('if-empty', null, InputOption::VALUE_NONE, 'lock if there are no pending locks for the same user')
-            ->addOption('ouuid', null, InputOption::VALUE_OPTIONAL, 'lock a specific ouuid', null)
+            ->addArgument(self::ARGUMENT_CONTENT_TYPE, InputArgument::REQUIRED, 'content type to recompute')
+            ->addArgument(self::ARGUMENT_TIME, InputArgument::REQUIRED, 'lock until (+1day, +5min, now)')
+            ->addOption(self::OPTION_QUERY, null, InputOption::VALUE_OPTIONAL, 'ES query', null)
```
Comment (theus77) [inline src/Command/LockCommand.php:58]:
```suggestion
            ->addOption(self::OPTION_QUERY, null, InputOption::VALUE_OPTIONAL, 'ES query', '{}')
```
I suggest to always do a queryin order to simplify the code

File: src/Command/LockCommand.php [hunk 3]
```diff
@@ -37,64 +53,121 @@ protected function configure(): void
...
 
-        if (0 === $rows) {
-            $io->error('no revisions locked, try force?');
+                return -1;
+            }
+
+            $revisionCount = 0;
+            foreach ($this->searchDocuments($search) as $document) {
+                $this->io->comment($document->getId());
+                $revisionCount += $this->revisionRepository->lockRevisions($this->contentType, $this->until, $this->by, $this->force, $document->getId());
```
Comment (theus77) [inline src/Command/LockCommand.php:136]: Maybe group the lock by 1000 ouuids ?

**Review by theus77 — APPROVED** [review]
Good work. Just left some ideas to think about.

**Review by dameert — APPROVED** [review]

File: src/Command/LockCommand.php (hunk 3 above)
Comment (coppee) [reply src/Command/LockCommand.php:136]: Seems a good idea but I'm not really sure how to do that properly....

Comment (theus77) [reply src/Command/LockCommand.php:136]: In the searchDocuments function yields an array of documents at each scroll instead of one document at the time.

**Review by theus77 — APPROVED** [review]

**Pull request merged**